### PR TITLE
feat: configurable windows os version

### PR DESCRIPTION
**Pull request by mweibel**
Allows to configure the windows version used for argoexec (see #5376). This allows us to easily build a version of the argoexec docker image without the need to adjust the Dockerfile itself (instead just pass `--build-arg IMAGE_OS_VERSION=1909`). 

This changes additionally the golang version used to the latest 1.13 version, as at least 1.13.8 is needed to fix an issue with windows (https://github.com/golang/go/issues?q=milestone%3AGo1.13.8+label%3ACherryPickApproved, https://github.com/golang/go/issues/36575). 

The `IMAGE_OS` and `IMAGE_ARCH` build args were unused and hence I removed them. 

possibly @lippertmarkus to review? :) 

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

### CODE REVIEW

**Review by lippertmarkus — APPROVED** [review]

File: Dockerfile.windows [hunk 1]
```diff
@@ -3,14 +3,14 @@
 # Initial stage which pulls prepares build dependencies and CLI tooling we need for our final image
 # Also used as the image in CI jobs so needs all dependencies
 ####################################################################################################
+
+ARG IMAGE_OS_VERSION=1809
+
 # had issues with official golange image for windows so I'm using plain servercore
 FROM mcr.microsoft.com/windows/servercore:ltsc2019 as builder
-ENV GOLANG_VERSION=1.13.4
+ENV GOLANG_VERSION=1.13.15
```
Comment (alexec) [inline Dockerfile.windows:11]: I wonder if this should be set to 1.15.7 like we use elsewhere?

Comment (mweibel) [reply Dockerfile.windows:11]: `release-2.12` where I based the branch on used 1.13.4 also for the normal `Dockerfile`. Wasn't sure whether I should upgrade to a newer major version if the standard one uses the same 1.13 release.. 

Of course that probably means forward porting to the current master might be a little difficult? Not sure how that usually works in this project though :)

**Review by simster7 — APPROVED** [review]

**Review by alexec — CHANGES_REQUESTED** [review]
This PR is based off `release-2.12`, but it must be based of `master`. Please update

**Comment by mweibel** [comment]
Will do @alexec - might take me another week as I'm out of office currently.

**Comment by mweibel** [comment]
d'oh - forgot you can't change the target branch in GitHub. Created #5816 with the new target branch.

**Pull request closed without merging**